### ISSUE TITLE
 calculate constants that need to be only calculated once per simulation, only once instead of Nbinaries times (to speed up code)

### ISSUE THREAD

**Issue by FloorBroekgaarden** [TeamCOMPAS/COMPAS#37]
at the moment, some constants are calculated for each binary, whereas they only have to be calculated once per simulation (the constants are the same for all binaries, but depend on OPTIONS in `pythonSubmit.py`. 

An example: `m_baryonicMassOfMaximumNeutronStarMass` is now calculated for each binary in `BaseStar.cpp`; whereas it should only be calculated once for all N binaries.

From `BaseStar.cpp`
`
    m_baryonicMassOfMaximumNeutronStarMass       = (0.075 * OPTIONS ->MaximumNeutronStarMass() * OPTIONS->MaximumNeutronStarMass()) + OPTIONS->MaximumNeutronStarMass();
    // calculate only once for entire simulation of N binaries in the future. 
}
`

In Jeff's words: 
we could enhance the code with maybe a global "constants" singleton 
"CONSTANTS->" object (like OPTIONS->) - we could incorporate constants.h into it, as well as anything that we only want to calculate once per run.

**Comment by ilyamandel** [comment]
@jeffriley -- this reminded me of the LOG10(Z) discussion... ;-) 

The real question is whether we want to do this on a reasonable timescale or not.  If not, I am tempted to assign the "won't fix" label and close.  Interested future developers with more time can always sort issues (open OR closed) on "won't fix" labels...

**Comment by jeffriley** [comment]
I think this would be a good thing to do.  I have more time now, and I will have even more time in three months...

**Comment by ilyamandel** [comment]
Keeping this alive for @jeffriley . :)

**Comment by ilyamandel** [comment]
@jeffriley -- gentle ping on whether you still want this to stay alive (I think it's a pretty low priority, so wouldn't mind closing it -- if the work is non-trivial, the cost savings are unlikely to be significant enough to make it worthwhile).

**Comment by jeffriley** [comment]
@ilyamandel I will implement this one day - I do think it is worthwhile, both from a performance perspective, and for clarity.  But happy to close this now and revisit later when I have some time.
